### ISSUE TITLE
`elidable_lifetime_names` incorrectly warns about a required lifetime

### ISSUE THREAD

**Issue by sdht0** [rust-lang/rust-clippy#15642]
### Summary

From https://github.com/kuzudb/kuzu/blob/0888644db324122ccbf63664fec07e22e95c1f0a/tools/rust_api/src/ffi.rs#L174

### Lint Name

elidable_lifetime_names

### Reproducer

I tried this code:

```rust
#[cxx::bridge]
pub(crate) mod ffi {
 unsafe extern "C++" {
        type Connection<'db>;
         ...
  }
}
```

I saw this happen:

```
warning: the following explicit lifetimes could be elided: 'db
   --> src/ffi.rs:174:25
    |
174 |         type Connection<'db>;
    |       

error[E0637]: `'_` cannot be used here
   --> src/ffi.rs:174:25
    |
174 |         type Connection<'_>;
    |                         ^^ `'_` is a reserved lifetime name

```

### Version

```text
rustc 1.89.0 (29483883e 2025-08-04)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: aarch64-unknown-linux-gnu
release: 1.89.0
LLVM version: 20.1.7
```
